internal/core/data/io.go

```diff
@@ -54,10 +55,12 @@ func (jsonReader) Read(reader io.Reader, ctx *context.Context) (models.Event, er
 }
 
 // cborReader handles unmarshaling of a CBOR request body payload
-type cborReader struct{}
+type cborReader struct {
+	configuration *config.ConfigurationStruct
+}
```

Comment:
I struggled with whether or not to provide a factory function to explicitly populate and return a cborReader instance.  On one hand, "consistency."  On the other, this struct is package private.  Then again, the package currently contains pretty much the entirety of core-data's implementation.  

I'd err on the side of consistency and recommend you create a factory function that takes the explicit dependencies, populates a cborReader instance with them, and returns the cborReader instance.

Reply:
There is a factory function for creating Reader instances:

https://github.com/edgexfoundry/edgex-go/blob/41e418c1fbeacadce475f5c89f03f5ec7ee74cf7/internal/core/data/io.go#L91-L102

Are you suggesting making another factory method?

Reply:
Reader is the contract, I think.  cborReader is an implementation of that contract.  What I'm after is avoiding creating a cborReader instance directly as a struct.  If I recall, we're doing that somewhere in the tests.